src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,266 @@
+      return;
+    }
+    this.addEventListener('keyup', this.#keyupListener, {once: true});
+  }
+
+  enable() {
+    this.addEventListener('click', this.#clickListener);
+    this.addEventListener('keydown', this.#keydownListener);
+    this.setAttribute('tabindex', 0);
+  }
+
+  disable() {
+    this.removeEventListener('click', this.#clickListener);
+    this.removeEventListener('keyup', this.#keyupListener);
+    this.removeAttribute('tabindex');
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === MediaUIAttributes.MEDIA_CONTROLLER) {
+      if (oldValue) {
+        const mediaControllerEl = document.getElementById(oldValue);
+        mediaControllerEl?.unassociateElement?.(this);
+      }
+      if (newValue) {
+        const mediaControllerEl = document.getElementById(newValue);
+        mediaControllerEl?.associateElement?.(this);
+      }
+    } else if (attrName === 'disabled' && newValue !== oldValue) {
+      if (newValue == null) {
+        this.enable();
+      } else {
+        this.disable();
+      }
+    }
+  }
+
+  connectedCallback() {
+    if (!this.hasAttribute('disabled')) {
+      this.enable();
+    }
+
+    this.setAttribute('role', 'listbox');
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.associateElement?.(this);
+    }
+  }
+
+  disconnectedCallback() {
+    this.disable();
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.unassociateElement?.(this);
+    }
+  }
+
+  get keysUsed() {
+    return ['Enter', ' ', 'ArrowDown', 'ArrowUp', 'Home', 'End'];
+  }
+
+  #getItem(e) {
+    if (e.target === this) return;
+
+    const composedPath = e.composedPath();
+    const slotIndex = composedPath.findIndex(el => el.nodeName === 'SLOT');
+
+    return composedPath[slotIndex - 1];
+  }
+
+  handleSelection(e) {
+    const item = this.#getItem(e);
+
+    if (!item) return;
+
+    const selected = item.getAttribute('aria-selected') === 'true';
+
+    const slot = this.shadowRoot.querySelector('slot');
+
+    if (this.getAttribute('aria-multiselectable') !== 'true') {
+      slot.assignedElements().forEach(el => el.removeAttribute('aria-selected'));
+    }
+
+    if (selected) {
+      item.removeAttribute('aria-selected');
+    } else {
+      item.setAttribute('aria-selected', 'true');
+    }
+  }
+
+  handleMovement(e) {
+    const { key } = e;
+
+    let currentOption = this.#getItem(e);
+    if (!currentOption) {
+      const slot = this.shadowRoot.querySelector('slot');
+      currentOption = slot.assignedElements()
+        .filter(el => el.getAttribute('tabindex') === '0')[0];
```

Comment:
could this be undefined? might need `...)?.[0]`

Reply:
Theoretically possible, but I think in practice unlikely.
However, yeah, we should still probably do a null check.